clang/lib/CodeGen/CodeGenModule.h

```diff
@@ -587,6 +590,12 @@ class CodeGenModule : public CodeGenTypeCache {
     return *CUDARuntime;
   }
 
+  /// Return a reference to the configured SYCL runtime.
+  CGSYCLRuntime &getSYCLRuntime() {
+    assert(SYCLRuntime != nullptr);
```

Comment:
Would a magic-static be better here?  You already have the CGM available, and that way you don't have to use the unique pointer at all.  I think I would prefer something like that instead.

Reply:
Please see earlier comments - I'm just following the general design. While I agree that magic static could possibly be used here, but it seems more important to have all runtimes follow the same design pattern for code clarity and maintenance. Magic static could be introduced for all runtimes at once.